SetupDataPkg/Tools/KnobService.py

```diff
@@ -147,6 +147,24 @@ def get_spdx_header(header, efi_type):
     spdx += get_line_ending(efi_type)
     return spdx
 
+# Check schema for header references and extract #include directives
+def get_schema_header_references(schema, efi_type):
+    headerSet = set()
+    for enum in schema.enums:
+        if enum.headerRef != "":
+            headerSet.add(enum.headerRef.strip())
+    for struct in schema.structs:
+        if struct.headerRef != "":
+            headerSet.add(struct.headerRef.strip())
+    if headerSet:
+        headers = "/* Schema-referenced headers */" + get_line_ending(efi_type)
+        for include in headerSet:
+            headers += '#include "{}"'.format(include)

```

Comment:
Bonus would be if we could track the set of type names that are intended to be included. i.e. so the output looks like:
```c
/* Schema-referenced headers */
#include "bar.h" // for bar_options_t
#include "feature.h" // for feature_config_t, fooenum
/*--------------------------*/
```

This would be helpful for readers to find the type definitions, and also for debugging the scripts/xml files if the expected types don't end up getting included correctly.

Reply:
Ok to assume long lists for potentially large headers are acceptable?

Reply:
Each definition block for both structs and enums that is defined optionally via a `headerRef` is now comment-tagged with the reference it comes from.